widget/embedded/src/components/SwapDetails/SwapDetails.tsx

```diff
@@ -0,0 +1,495 @@
+import { numberToString } from '../../utils/numbers';
+import { getPendingSwaps } from '../../utils/queue';
+import {
+  getLastConvertedTokenInFailedSwap,
+  getPercentageChange,
+  getSwapMessages,
+  shouldRetrySwap,
+} from '../../utils/swap';
+import { getSwapDate } from '../../utils/time';
+import { getConciseAddress, getUsdPrice } from '../../utils/wallets';
+import { Layout } from '../Layout';
+
+import {
+  AMOUNT_DECIMALS,
+  getPriceImpactWarningLevel,
+  getSteps,
+  getStepState,
+  MIN_DECIMALS,
+  PERCENT_DECIMALS,
+  RESET_INTERVAL,
+  SECONDS,
+} from './SwapDetails.helpers';
+import { ModalSwapDetails } from './SwapDetails.Modal';
+import { SwapDetailsPlaceholder } from './SwapDetails.Placeholder';
+import {
+  Container,
+  HeaderDetails,
+  Separator,
+  StepsList,
+  StyledButton,
+} from './SwapDetails.styles';
+
+export function SwapDetails() {
+  const requestId = useUiStore.use.selectedSwapRequestId();
+  const { tokens } = useMetaStore.use.meta();
+  const { canSwitchNetworkTo, connect, getWalletInfo } = useWallets();
+  const retry = useBestRouteStore.use.retry();
+  const navigate = useNavigate();
+
+  const { navigateBackFrom } = useNavigateBack();
+
+  const [_, handleCopy] = useCopyToClipboard(RESET_INTERVAL);
+  const { manager, state } = useManager();
+  const [modalState, setModalState] = useState<ModalState>(null);
+  const [showCompletedModal, setShowCompletedModal] = useState<
+    'success' | 'failed' | null
+  >(null);
+
+  const pendingSwaps = getPendingSwaps(manager);
+  const selectedSwap = pendingSwaps.find(
+    ({ swap }) => swap.requestId === requestId
+  );
+
+  const onCancel = () => {
+    if (selectedSwap?.id) {
+      const swap = manager?.get(selectedSwap.id);
+      if (swap) {
+        cancelSwap(swap);
+        setModalState(null);
+      }
+    }
+  };
+
+  const deleteFromQueue = async () => {
+    if (selectedSwap?.id) {
+      try {
+        await manager?.deleteQueue(selectedSwap.id);
+        navigateBackFrom(navigationRoutes.swapDetails);
+      } catch (e) {
+        console.log(e);
+      }
+    }
+  };
+  const swap = selectedSwap?.swap;
+  const loading = !state.loadedFromPersistor;
+
+  const currentStep = swap ? getCurrentStep(swap) : ({} as PendingSwapStep);
+
+  const currentStepNetworkStatus = currentStep?.networkStatus;
+
+  useEffect(() => {
+    if (swap?.status === 'success' || swap?.status === 'failed') {
+      setShowCompletedModal(swap?.status);
+    }
+  }, [swap?.status]);
+
+  useEffect(() => {
+    if (swap && showSwitchNetwork) {
+      setModalState(PendingSwapNetworkStatus.WaitingForNetworkChange);
+    } else if (
+      currentStepNetworkStatus ===
+        PendingSwapNetworkStatus.WaitingForConnectingWallet ||
+      currentStepNetworkStatus === PendingSwapNetworkStatus.NetworkChanged
+    ) {
+      setModalState(currentStepNetworkStatus);
+    } else {
+      setModalState(null);
+    }
+  }, [currentStepNetworkStatus]);
+
+  if (!swap) {
```

Comment:
Is there any reason to put your rendering code here in the middle of logic codes?

Reply:
fixed.

Reply:
fixed.